tool-execution/run_pmd.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+## This script sets up, and runs PMD on an Ubuntu 20.04 or MacOS 11 machine.
+###########
+## SETUP ##
+###########
+
+function configure_pmd() {
+    # create pmd dir if not exists
+    mkdir -p "$1"
+
+    ## create config.xml file
+    echo "<?xml version='1.0'?>
+    <ruleset name='Custom Rule for Cyclomatic Complexity'>
+        <description>Cyclomatic complexity</description>
+        <rule ref='category/java/design.xml/CyclomaticComplexity'>
+            <properties>
+                <property name='methodReportLevel' value='1' />
+            </properties>
+        </rule>
+    </ruleset>" > "$1"/cyclomatic_complex_rules.xml
+}
+
+PMD_DIR=../pmd
+configure_pmd "${PMD_DIR}"
+
+## checks if the environment variable pmd is set
+if ! command -v pmd &> /dev/null ; then
+    echo "cannot find pmd on this machine."
+    echo "This script will run tool-execution/pmd-setup.sh now if you are on an Ubuntu 20.04 machine or Mac OS 11+. Otherwise, read the instructions in that script on how to adapt it for other platforms, then follow them for yours."
+    ## run pmd-setup.sh to setup pmd
+    ./pmd_setup.sh
```

Comment:
This needs to be checked for failure: that is, after you run it, you should make sure that pmd is actually installed. If it is not, you should exit this script. (this would happen if e.g. you were on the wrong platform).